homeassistant/components/azure_servicebus/notify.py

```diff
@@ -0,0 +1,83 @@
+"""Support for azure service bus notification."""
+import json
+import logging
+
+from azure.servicebus.aio import Message, ServiceBusClient
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    ATTR_MESSAGE,
+    ATTR_TARGET,
+    ATTR_TITLE,
+    PLATFORM_SCHEMA,
+    BaseNotificationService,
+)
+from homeassistant.const import CONTENT_TYPE_JSON
+import homeassistant.helpers.config_validation as cv
+
+CONF_CONNECTION_STRING = "connection_string"
+CONF_QUEUE_NAME = "queue"
+CONF_TOPIC_NAME = "topic"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_CONNECTION_STRING): cv.string,
+        vol.Exclusive(
+            CONF_QUEUE_NAME, "output", "Can only send to a queue or a topic."
+        ): cv.string,
+        vol.Exclusive(
+            CONF_TOPIC_NAME, "output", "Can only send to a queue or a topic."
+        ): cv.string,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the notification service."""
+    connection_string = config[CONF_CONNECTION_STRING]
+    queue_name = config.get(CONF_QUEUE_NAME)
+    topic_name = config.get(CONF_TOPIC_NAME)
+
+    servicebus = ServiceBusClient.from_connection_string(
+        connection_string, loop=hass.loop
+    )
+
+    if queue_name:
+        client = servicebus.get_queue(queue_name)
+    elif topic_name:
+        client = servicebus.get_topic(topic_name)
+    else:
+        _LOGGER.error(
+            "Could not find a topic or queue from the configuration. Please provide one!"
+        )
+        return None
+
+    return ServiceBusNotificationService(client)
+
+
+class ServiceBusNotificationService(BaseNotificationService):
+    """Implement the notification service for the service bus service."""
+
+    def __init__(self, client):
+        """Initialize the service."""
+        self._client = client
+
+    async def async_send_message(self, message, **kwargs):
+        """Send a message."""
+        dto = {ATTR_MESSAGE: message}
+
+        if ATTR_TITLE in kwargs:
+            dto[ATTR_TITLE] = kwargs[ATTR_TITLE]
```

Comment:
Are the keys in the dto defined by the Azure api, or are they arbitrary? If they are Azure api specific we should not use the same constants as in the home assistant api. That could break the integration if we change the home assistant constants.

Reply:
Every field into the dto is arbitrary, so it shouldn't make a difference if the names changes, but you have a good point that if the constants change it would also make the output change. 
Have added separate constants for the output to servicebus, so if the internals change it doesn't break anything anyone has built on top of this.